modules/vpc_association/variables.tf

```
@@ -0,0 +1,14 @@
+variable "vpc_id" {
+  type = object({
+    vpc_id = optional(string)
+  })
+  description = "VPC ID"
+  default = {
+    vpc_id = null
+  }
+}
```

Can simplify here also, making vpc_id required:

```suggestion
variable "vpc_id" {
  description = "VPC ID"
  type = string
}
```